src/07-even-more-loops.js

```diff
@@ -28,7 +35,16 @@ function shoutForLoop(array) {}
  *  shoutWhileLoop([ "A", "Very", "Happy", "Array" ]);
  *  //> [ "A!", "Very!", "Happy!", "Array!" ];
  */
-function shoutWhileLoop(array) {}
+function shoutWhileLoop(array) {
+
+  let count = 1;
+  while (count <= array.length) {
+    array.push(count++)
+    console.log(array);
+   }
+   
+   return array["!"];
```

Comment:
line 42 is doing some funny stuff - first it's iterating the value of ` count `.  Second it is pushing that value into your ` array `.  Lin 46 is trying to return the ` array ` at the index of ` ! `.  Remember we can only index arrays by number so 46 will throw an error.  Secondly while iterating through an array we need to determine _what our index_ is and what _the item in the array at the index is_.  These are two crucial concepts to understanding loops